pandas/core/window.py

```diff
@@ -671,14 +679,24 @@ def _apply_window(self, mean=True, **kwargs):
         window = self._prep_window(**kwargs)
         center = self.center
 
-        blocks, obj, index = self._create_blocks()
+        blocks_dict, obj, index = self._create_blocks()
+        dtypes = blocks_dict.keys()
+        blocks = blocks_dict.values()
+
         results = []
-        for b in blocks:
+        exclude = []
+        for dtype in list(dtypes):
+            b = blocks_dict[dtype]
             try:
                 values = self._prep_values(b.values)
-            except TypeError:
-                results.append(b.values.copy())
-                continue
+
+            except (TypeError, NotImplementedError):
+                if isinstance(obj, ABCDataFrame):
+                    exclude.extend(b.columns)
+                    del blocks_dict[dtype]
```

Comment:
why are you del here?

Reply:
As you stated the order for dictionary differs in each run. So iteration order for `blocks` differs.

So in case of a DataFrame with columns `["A", "B"]` with types `[int, str]` when iteration starts with "B" then:

`results = [values_of_A]` but `blocks = [block_for_B, block_for_A]`. There seems to be a mismatch  [here](https://github.com/pandas-dev/pandas/blob/ad9b5e2aeaaa32ced72ef2be65d7067c3bce0345/pandas/core/window.py#L262) (values_of_A and block_for_B are iterated together). So I went with removing `block_for_B` which was the first thing came to my mind.